opennms-provision/opennms-detectorclient-rpc/src/main/java/org/opennms/netmgt/provision/detector/client/rpc/DetectorClientRpcModule.java

```diff
@@ -71,11 +71,8 @@ public DetectorClientRpcModule() {
     public CompletableFuture<DetectorResponseDTO> execute(DetectorRequestDTO request) {
         String className = request.getClassName();
         Map<String, String> attributes = request.getAttributeMap();
-        ServiceDetector detector = serviceDetectorRegistry.getDetectorByClassName(className, attributes);
-        if (detector == null) {
-            throw new IllegalArgumentException("No detector found with class name '" + className + "'.");
-        }
-        return detectService(detector, request);
+        return serviceDetectorRegistry.getDetectorFutureByClassName(className, attributes)
```

Comment:
This future may never resolve if a detector doesn't exist for the given `className` right?